compiler-core/src/parse/error.rs

```diff
@@ -232,6 +232,14 @@ utf16_codepoint, utf32_codepoint, signed, unsigned, big, little, native, size, u
                     "See: https://tour.gleam.run/flow-control/case-expressions/".into(),
                 ],
             ),
+            ParseErrorType::ListPatternAfterSpread => (
+            "Invalid pattern", vec![
+                "I was expecting the end of the list.".into(),
+                "The spread pattern precesing this consumes the remaining elements of the list".into(),
+                "therefore there is nothing left to match.".into()
```

Comment:
Let's have the message explain why it is that you cannot append to a list.

Reply:
I've added the explanation from `ListSpreadFollowedByElements` let me know if you anything more specific in mind.